lib/coll_8_api/errors/response_error.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Coll8Api
+  module Errors
+    class ResponseError < StandardError
+      attr_reader :payload, :status, :body
+
+      def initialize(payload:, status:, body:)
+        @payload = payload
+        @status = status
+        @body = body
+
+        super(build_message)
+      end
+
+      private
+
+      def build_message
+        return status unless body.is_a?(Hash)
+
+        body.dig(:reasons, 0, :msg) || status
```

Comment:
Are we certain that Coll8 returns this erroneous response structure?

Reply:
Will return simple status until with have concrete examples of the error messaging.